### PR TITLE
Excercise Equipment is now craftable

### PR DESCRIPTION
**Pull request by flowercuco**
## About The Pull Request

Imagine if you will a humble chaplain who wants nothing more than for all of the spiritual folk on the station to get as massive gains as they can, after finding that they can not just make more exercise equipment and that the station does not have any in public places, they go annoy security enough to get into permabrig only to find out that they cant even unwrench the equipment and move it to the church!!! 

NOT ANYMORE!!!

jS2aBMBa0B
crafting recipies

dreamseeker_O4BgBRsFa8
demonstrating unwrenching and wrenching equipment

dreamseeker_hCFQJZdzoS
crafting a punching bag and punching it

Now you can craft as much exercise equipment as you want! May everyone on the station get as strong as possible and not just prisoners. 

Also I changed the message that plays when you try to use exercise equipment someone else is using into a balloon alert.
dreamseeker_PwNesmcR1f

## Why It's Good For The Game

Access to exercise equipment on some maps is limited to static positions and is currently mostly only for prisoners as every map does not have public exercise equipment. Expanding the access means that you can have a Drill Sargent Head of Security or Captain who commands people use these or allows a psychologist to prescribe healthy exercise habits to their patients. 

I think having the potential for exercise equipment on every map is more fun and also if prisoners get their hands on tools they should be allowed to mess with these to annoy security or aid in their escape.

## Changelog
:cl:
add: the punching bag, bench press, and chest press are all able to be crafted and unanchored.
add: crafting recipes for the above
qol: changed a chat message into a balloon alert
qol: adds screentips to equipment (thanks for suggesting i do this mothblocks!)
/:cl:

### CODE REVIEW

**Comment by Farquaar** [comment]
Can't wait to set up a clown-gym in the bar!

**Comment by Jackal-boop** [comment]
> Can't wait to set up a clown-gym in the bar!
Bonus points If you bug the bartender for energy drinks

**Comment by Wallemations** [comment]
Unsecuring a punching bag and it remaining on the roof is a cursed visual

File: code/game/objects/structures/gym.dm [hunk 1]
```diff
@@ -20,6 +20,31 @@
 		L.add_mood_event("exercise", /datum/mood_event/exercise)
 		L.apply_status_effect(/datum/status_effect/exercised)
 
+/obj/structure/punching_bag/wrench_act(mob/living/user, obj/item/tool)
+	tool.play_tool_sound(src)
+	if(anchored)
+		balloon_alert(user, ("unsecuring"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:26]:
```suggestion
		balloon_alert(user, "unsecured")
```

File: code/game/objects/structures/gym.dm [hunk 2]
```diff
@@ -20,6 +20,31 @@
...
+		anchored = FALSE
+	else
+		balloon_alert(user, ("securing"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:29]:
```suggestion
		balloon_alert(user, "secured")
```

File: code/game/objects/structures/gym.dm [hunk 3]
```diff
@@ -20,6 +20,31 @@
...
+		anchored = TRUE
+	return TRUE
+
+/obj/structure/punching_bag/crowbar_act(mob/living/user, obj/item/tool)
+	if(anchored)
+		balloon_alert(user, ("unsecure first"))
+		return FALSE
+	tool.play_tool_sound(src)
+	balloon_alert(user, ("deconstructing"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:38]:
```suggestion
	balloon_alert(user, "deconstructing...")
```

... = pending action

File: code/game/objects/structures/gym.dm [hunk 4]
```diff
@@ -20,6 +20,31 @@
...
+		balloon_alert(user, ("unsecuring"))
+		anchored = FALSE
+	else
+		balloon_alert(user, ("securing"))
+		anchored = TRUE
+	return TRUE
+
+/obj/structure/punching_bag/crowbar_act(mob/living/user, obj/item/tool)
+	if(anchored)
+		balloon_alert(user, ("unsecure first"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:35]:
```suggestion
		balloon_alert(user, "unsecure first!")
```

! == failed

File: code/game/objects/structures/gym.dm [hunk 5]
```diff
@@ -47,7 +72,7 @@
 	if(!user.canUseTopic(src, be_close = TRUE, no_dexterity = FALSE, no_tk = TRUE))
 		return
 	if(obj_flags & IN_USE)
-		to_chat(user, span_warning("It's already in use - wait a bit!"))
+		balloon_alert(user, ("Wait your turn!"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:75]:
```suggestion
		balloon_alert(user, "wait your turn!")
```

File: code/game/objects/structures/gym.dm [hunk 6]
```diff
@@ -68,6 +93,31 @@
 		to_chat(user, finishmessage)
 		user.apply_status_effect(/datum/status_effect/exercised)
 
+/obj/structure/weightmachine/wrench_act(mob/living/user, obj/item/tool)
+	tool.play_tool_sound(src)
+	if(anchored)
+		balloon_alert(user, ("unsecuring"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:99]:
```suggestion
		balloon_alert(user, "unsecured")
```

File: code/game/objects/structures/gym.dm [hunk 7]
```diff
@@ -68,6 +93,31 @@
...
+		anchored = FALSE
+	else
+		balloon_alert(user, ("securing"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:102]:
```suggestion
		balloon_alert(user, "secured")
```

File: code/game/objects/structures/gym.dm [hunk 8]
```diff
@@ -68,6 +93,31 @@
...
+		anchored = TRUE
+	return TRUE
+
+/obj/structure/weightmachine/crowbar_act(mob/living/user, obj/item/tool)
+	if(anchored)
+		balloon_alert(user, ("unsecure first"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:108]:
```suggestion
		balloon_alert(user, "unsecure first!")
```

File: code/game/objects/structures/gym.dm [hunk 9]
```diff
@@ -68,6 +93,31 @@
...
+		return FALSE
+	tool.play_tool_sound(src)
+	balloon_alert(user, ("deconstructing"))
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:111]:
```suggestion
	balloon_alert(user, "deconstructing...")
```

File: code/game/objects/structures/gym.dm [hunk 10]
```diff
@@ -20,6 +20,31 @@
 		L.add_mood_event("exercise", /datum/mood_event/exercise)
 		L.apply_status_effect(/datum/status_effect/exercised)
 
+/obj/structure/punching_bag/wrench_act(mob/living/user, obj/item/tool)
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:23]: All of these should support `_act_secondary` as well--we want to formalize *de*construction being on right-click, not left-click.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
You should also add contextual screentips for tools. Let me know if you need help on that

File: code/game/objects/structures/gym.dm [hunk 11]
```diff
@@ -20,22 +39,22 @@
 		L.add_mood_event("exercise", /datum/mood_event/exercise)
 		L.apply_status_effect(/datum/status_effect/exercised)
 
-/obj/structure/punching_bag/wrench_act(mob/living/user, obj/item/tool)
+/obj/structure/punching_bag/wrench_act_secondary(mob/living/user, obj/item/tool)
 	tool.play_tool_sound(src)
 	if(anchored)
-		balloon_alert(user, ("unsecuring"))
+		balloon_alert(user, "unsecure")
```
Comment (Mothblocks) [inline code/game/objects/structures/gym.dm:45]: unsecured--past tense, same with tha others

**Comment by flowercuco** [comment]
thank you for the help! i've added the screentips for barehand and right mouse with the proper tools and changed the wrench stuff and crowbar stuff to right clicking. let me know if i messed up here haha

**Review by Mothblocks — APPROVED** [review]
Cute pfp btw

**Comment by flowercuco** [comment]
thank you very much!

**Comment by WEWLAD2222** [comment]
Hey buddy, I think you got the wrong gym. Cargonia's Crossfit Corner is three doors down.

**Pull request merged**